### PR TITLE
[MAINTENANCE] Move `get_context` out of utils

### PR DESCRIPTION
**Pull request by cdkini**
Move logic into separate `context_factory.py` file (can follow up this PR with a refactor into a factory class)

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/util.py [hunk 1]
```diff
@@ -121,6 +116,16 @@ def __delitem__(self, key: str):
...
 
 
+def get_context(*args, **kwargs) -> AbstractDataContext:
+    # deprecated-v0.17.12
+    warnings.warn(
+        "Importing `get_context` from `great_expectations.util` is deprecated as of v0.17.12 and will be removed in v0.20. Please import from the top-level `great_expectations` module.",
+        DeprecationWarning,
+    )
+
+    return context_factory(*args, **kwargs)
```
Comment (cdkini) [inline great_expectations/util.py:126]: Is this the best way to do this?

Comment (cdkini) [reply great_expectations/util.py:126]: I don't want users poking around util files

**Review by Kilo59 — APPROVED** [review]

File: great_expectations/util.py (hunk 1 above)
Comment (Kilo59) [reply great_expectations/util.py:126]: Yeah, this makes sense.

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM

**Pull request merged**